pyanaconda/ui/gui/spokes/source.py

```diff
@@ -403,6 +405,14 @@ def apply(self):
                 checkmount=False)
         self.clear_info()
 
+    def _repo_id_generator(self):
+        """ Repository id numbers generator
+        """
+        repo_id = 0
+        while(True):
+            yield repo_id
+            repo_id += 1
+
```

Comment:
This should be a more generic thing not taking ``self`` and defined as ``iutil.id_generator``.